fix: reject reconnecting agents with different resource pool configuration (#9815)
determined-ai/determined: docs/get-started/architecture/introduction.rst

:ref:`agent configuration <agent-config-reference>` to specify which resource pool the agent should
join.

> Note that to change an agent's assigned resource_pool after it has already joined one, you need to

Reviewer: ```suggestion
To change the resource pool an agent is assigned to after it has already joined one, you need to
```

> update the :ref:`agent configuration <agent-config-reference>`. Make sure to drain the agents

Reviewer: ```suggestion
update the :ref:`agent's configuration <agent-config-reference>`. Before making this change, ensure the agents are properly drained. Once the configuration is updated, restart the agent to connect it to the new resource pool.
```

> properly before modifying the resource_pool. After making the changes, restart the agent to connect

Reviewer: ```suggestion
```

> it to the new resource pool.

Reviewer: ```suggestion
```
